File: Android_app/pioneer/src/main/java/com/ABC/pioneer/sensor/Sensor.java
//  Copyright 2020 VMware, Inc.
//  SPDX-License-Identifier: Apache-2.0
//

package com.ABC.pioneer.sensor;

///用于检测和跟踪各种疾病传播媒介的传感器
public interface Sensor {
    /// 添加delegate以响应传感器事件。
    void add(SensorDelegate delegate);

    /// 开始检测。
    void start();

    /// 停止检测。
    void stop();
}

File: IOS_app/Pioneer_ios/Pioneer Framework/Pioneer/Sensor/Sensor.swift
//
//  Sensor.swift
//  Pioneer
//
//  Created by Beh on 2021/5/22.
//

import Foundation

///用于综合检测和跟踪各种疾病传播媒介的传感器，例如位置传感器、蓝牙传感器等。
public protocol Sensor {
    ///添加用于响应传感器事件的委托。
    func add(delegate: SensorDelegate)
    
    func start()
    
    func stop()
}

